make_plant_parts_eia.py

```diff
@@ -869,26 +939,45 @@ def dedup_on_category(self, dedup_df, base_cols, category_name, sorter):
         dedup_df = dedup_df.sort_values(category_name)
         return dedup_df.drop_duplicates(subset=base_cols, keep='first')
 
-    def grab_qualifiers(self,
-                        part_df,
-                        record_name,
-                        id_cols,
-                        denorm_table,
-                        denorm_cols
-                        ):
+    def get_qualifiers(self,
+                       part_df,
+                       record_name,
+                       id_cols,
+                       denorm_table,
+                       denorm_cols
+                       ):
         """
-        Grab qualifier records.
-
-        For an individual compiled dataframe for each of the plant parts, we
-        want to add back in some non-data columns (qualifier columns). When
-        qualitative data is consistent for every record in a plant part, we
-        assign these catagoricals. If the records are not consistent, then
-        nothing is added.
+        Get qualifier records.
+
+        For an individual dataframe of one plant part, we typically have
+        identifying columns and aggregated data columns. The identifying
+        columns are only the identifying columns associated with that plant
+        part (i.e. for plant_unit (plant part) we have ('plant_id_eia' and
+        'unit_id_pudl')). Every plant part is cobbled together from generator
+        records, so each record in each part_df can be thought of as a
+        collection of generators.
+
+        Identifier and qualifier columns are the same columns; whether a column
+        is an identifier or a qualifier is a function of the plant part you're
+        considering. All the other columns which could be identifiers in the
+        context of other plant parrts (but aren't for this plant part) are
+        qualifiers.
+
+        This method takes a part_df and goes and checks whether or not the data
+        we are trying to grab from the record_name column is consistent across
+        every component genertor from each record.
+
+        When record_df is "operational_status", we are not going to check for
+        consistency; instead we will grab the highest level of operational
+        status that is associated with each records' component generators. The
+        order of operational status is defined within the method as:
+        'existing', 'proposed', then 'retired'.
```

Comment:
For example if a plant_unit is composed of two generators, and one of them is "existing" and another is "retired" the entire plant_unit will be considered "existing".